lib/foreman/http_proxy/net_http_extension.rb

```diff
@@ -0,0 +1,17 @@
+Net::HTTP.class_eval do
+  include Foreman::HTTPProxy
+  alias_method :orig_request, :request
+
+  def request(req, body = nil, &block)
+    host = URI.parse(@address).host
+    if proxy_http_request?(@proxy_address, host, @socket)
+      log_proxied_request(http_proxy, host)
+      @proxy_address = URI.parse(http_proxy)
+      http_proxied_rescue do
+        orig_request(req, body, &block)
+      end
+    else
+      orig_request(req, body, &block)
+    end
+  end
+end
```

Comment:
I have an impression that this does not actually work. Example:

```ruby
Net::HTTP.start(uri.host, uri.port) do |http|
  http.request(...)
end
```

The thing is that method `start` actually opens connection already which fails on networks which don't have route to the destination and method `request` will never be called as the application (or main request) errors out.

I am filing https://projects.theforeman.org/issues/27065 for this.